### DISCUSSION TITLE
Best practice to compare the results before and after code change

### DISCUSSION THREAD

**Discussion by liutiming** [ropensci/targets#305] [General]
Hi @wlandau , I remember our discussion of data version control and was thinking it might be relevant to this. I often get into situations where I want to compare the results before and after code change. I am wondering if you or anyone has a workflow for that?

**Comment by liutiming** [comment]
For reference, https://dvc.org/doc/start/data-versioning is sufficient for toggle between project status, but I have not found a way to compare without significant code duplication

**Reply by wlandau** [reply]
Thanks, looks pretty slick to use `dvc ...` for `_targets/` and `git ...` for code. Would you elaborate on the code duplication issue?

**Reply by liutiming** [reply]
Yea it is taking quite long for a large _targets though...

RE code duplication, if I have analysis a -> z, and I realise a -> b has a bug so I want to correct that. But I also want to see what was the impact of the bug on z, so in these cases I have to duplicate a2->z2 to compare z to z2 directly in targets.

One non-reproducible way to do it is to save z to a file and then compare that.

**Reply by wlandau** [reply]
Git branches solve the code duplication problem quite nicely, as long as they have informative names. Extensions to Git such as [VSCode Git Graph](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph) can help manage branches locally.

**Reply by liutiming** [reply]
thanks I will try it out!
